system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -73,7 +69,7 @@ int __timedwait_cp(volatile int *addr, int val,
 	    pthread_self()->cancelasync == PTHREAD_CANCEL_ASYNCHRONOUS) {
 		double sleepUntilTime = emscripten_get_now() + msecsToSleep;
 		do {
-			if (_pthread_isduecanceled(pthread_self())) {
+			if (pthread_self()->cancel) {
```

Comment:
As a followup we should replace `pthread_self` with `__pthread_self` in all our musl-internal code. 

Also, `pthread_t self = __pthread_self();` would probably make sense here.